pkg/volume/gce_pd/gce_util.go

```diff
@@ -109,7 +111,11 @@ func (gceutil *GCEDiskUtil) CreateVolume(c *gcePersistentDiskProvisioner) (strin
 			zonesPresent = true
 			configuredZones = v
 		case "replication-type":
-			replicationType = strings.ToLower(v)
+			if utilfeature.DefaultFeatureGate.Enabled(features.GCERegionalPersistentDisk) {
+				replicationType = strings.ToLower(v)
+			} else {
+				return "", 0, nil, "", fmt.Errorf("invalid option %q for volume plugin %s", k, c.plugin.GetPluginName())
```

Comment:
What if this `replication-type` is set to `none`? We wouldn't want to fail in that case right.

Why not just put a new check at line 124:

```
if replicationType == replicationTypeRegionalPD 
    && !utilfeature.DefaultFeatureGate.Enabled(features.GCERegionalPersistentDisk) {
  return "errorFeatureNotEnabled"
}
```

Then you can revert all the other changes below.

Reply:
Strictly speaking, `replication-type` doesn't exist before adding regional PD support, so my opinion is `replication-type = none` shouldn't be allowed either.